sdk/core/azure-core/azure/core/tracing/ext/opencensus_wrapper.py

```diff
@@ -0,0 +1,162 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+
+from azure.core.tracing import AbstractSpan
+from opencensus.trace import execution_context
+from opencensus.trace import tracer as tracer_module, Span
+from opencensus.trace.propagation import trace_context_http_header_format
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracers.noop_tracer import NoopTracer
+
+try:
+    from opencensus.ext.azure.trace_exporter import AzureExporter
+except:
+    AzureExporter = False
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Union
+
+
+class OpencensusSpanWrapper(AbstractSpan):
+    """Wraps a given Opencensus Span so that it implements azure.core.tracing.AbstractSpan"""
+
+    def __init__(self, span=None, name="parent_span"):
+        # type: (Span, str) -> None
+        """
+        If a span is not passed in, creates a new tracer. If the instrumentation key for Azure Exporter is given, will
+        configure the azure exporter else will just create a new tracer.
+
+        :param span: The opencensus span to wrap
+        :param name: The name of the opencensus span to create if a new span is needed
+        """
+        super(OpencensusSpanWrapper, self).__init__(span, name)
+        tracer = self.get_current_tracer()
+        self.was_created_by_azure_sdk = False
+        if span is None and (tracer is None or isinstance(tracer, NoopTracer)):
+            azure_exporter_instrumentation_key = os.environ.get(
+                "APPINSIGHTS_INSTRUMENTATIONKEY", None
+            )
+            prob = float(os.environ.get("AZURE_TRACING_SAMPLER", None) or 0)
+            if azure_exporter_instrumentation_key is not None and AzureExporter:
+                tracer = tracer_module.Tracer(
+                    exporter=AzureExporter(
+                        instrumentation_key=azure_exporter_instrumentation_key
+                    ),
+                    sampler=ProbabilitySampler(prob),
+                )
+            else:
+                tracer = tracer or tracer_module.Tracer(
+                    sampler=ProbabilitySampler(prob)
+                )
+            self.was_created_by_azure_sdk = True
+        span = span or tracer.span(name=name)
+
+        self.tracer = tracer
+        self._span_instance = span
+
+    @property
+    def span_instance(self):
+        # type: () -> Span
+        """
+        :return: The openencensus span that is being wrapped.
```

Comment:
~~openencensus~~ OpenCensus

Reply:
Done

Reply:
Did you push the commit?

Reply:
yes 
https://github.com/Azure/azure-sdk-for-python/pull/6264/files#diff-3a7fbaedc58e7956d36ab864e6fb3e4aR46